### ISSUE TITLE
pex_binary fails on jsonschema requirement with python 3 target

### ISSUE THREAD

**Issue by hugoduncan** [benley/bazel_rules_pex#9]
When targeting python 3.5, requiring jsonschema leads to `pex` trying to install `functools32` (and failing, since the package is python 2.7 only).

```
$ /usr/local/bin/python3 --version
Python 3.5.2
$ bazel help | head -1
                                                  [bazel release 0.3.0-homebrew]
$ mkdir repro
$ cd repro
$ cat > repro.py <<EOF
import jsonschema

if __name__ == '__main__':
   print('hello')
EOF
$ cat > BUILD <<EOF
pex_binary(
    name="repro",
    srcs=['repro.py'],
    main="repro.py",
    interpreter="/Usr/local/bin/python3",
    reqs=[
        "jsonschema==2.5.1"
    ]
 )
EOF
$ cd ..
$ bagel run //repro
...
**** Failed to install functools32-3.2.3-2. stderr:
This backport is for Python 2.7 only.
...
```

`jsonschema` declares its dependency as:

```
Requires-Dist: functools32; python_version=='2.7'
```

It looks like `pex` is not reading this information correctly.

This is with `bazel_rules_pex` 4882f2ddaf10f447754833643d1b266b05fce6ea.

**Comment by benley** [comment]
I think I know at least one way to fix this: it seems like if you make sure pex itself is executed using python3, it handles python3 things correctly.  I'm trying out a couple of approaches.

**Comment by mouadino** [comment]
I didn't see this one before reporting mine, but the issue is the same as this https://github.com/benley/bazel_rules_pex/issues/44 and I have a fix for it already.

**Comment by benley** [comment]
Sounds like this is fixed via #45!
